### PR TITLE
Handling of empty keys

### PR DESCRIPTION
**Pull request by monich**
Don't pass empty key to MGConfItem. Return the default value if the key is empty or has never been set.

### CODE REVIEW

File: src/configurationvalue.cpp [hunk 1]
```diff
@@ -51,29 +52,30 @@ QString ConfigurationValue::key() const
...
+    QVariant oldValue = value();
+    QString oldKey = key();
 
     delete mItem;
-    mItem = new MGConfItem(key, this);
-    connect(mItem, SIGNAL(valueChanged()), this, SIGNAL(valueChanged()));
-
-    emit keyChanged();
+    if (newKey.isEmpty()) {
+        mItem = NULL;
```
Comment (pvuorela) [inline src/configurationvalue.cpp]: 0 would be more common in C++ world. Also used in ctor.

File: src/configurationvalue.cpp [hunk 2]
```diff
@@ -51,29 +52,30 @@ QString ConfigurationValue::key() const
...
+    } else {
+        mItem = new MGConfItem(newKey, this);
+        connect(mItem, SIGNAL(valueChanged()), this, SIGNAL(valueChanged()));
+    }
 
-    // we deleted the old item, so we must emit ourselves in this case
-    if (oldValue != mItem->value(mDefaultValue))
+    // don't emit valueChanged and keyChanged unless absolutely necessary
+    if (oldKey != key())
+        emit keyChanged();
```
Comment (pvuorela) [inline src/configurationvalue.cpp]: Small optimization would be to have this check at the beginning, just returning if key didn't change.

**Comment by pvuorela** [comment]
Couple small comments, but looks good this way too.

**Comment by monich** [comment]
Updated the PR based on the review comments

**Comment by pvuorela** [comment]
LGTM

**Pull request merged**